documentation/puppet-api/v3/environment_classes.md

```diff
@@ -4,84 +4,121 @@ title: "Puppet Server: Puppet API: Environment Classes"
+
+Queries to the resource type API use cached class information per the configuration of the
+[`environment_timeout`][] setting, as set in the corresponding environment's
+`environment.conf` file. The environment classes API does not use the value of
+`environment_timeout` with respect to the data that it caches. Instead, only when the
+`environment-class-cache-enabled` setting in the `jruby-puppet` configuration section is
+set to `true`, the environment classes API uses HTTP [Etags][Etag] to represent specific
+versions of the class information and the Puppet Server [environment cache API][] as an
+explicit mechanism for marking an Etag as expired. See the
+[Headers and Caching Behavior](#headers-and-caching-behavior) section for more information
+about caching and invalidation of entries.
+
+### Uses typed values
+
+The environment classes API includes a `type`, if defined for a class parameter. For
+example, if the class parameter were defined as `String $some_str`, the `type` parameter
+would hold a value of `String`.
+
+### Provides default literal values
+
+For values that can be presented in pure JSON, the environment classes API provides a
+`default_literal` form of a class parameter's default value. For example, if an `Integer`
+type class parameter were defined in the manifest as having a default value of `3`, the
+`default_literal` element for the parameter will contain a JSON Number type of 3.
+
+### Lacks filters
+
+The environment classes API does not provide a way to filter the list of classes returned
+via use of a search string. The environment classes API returns information for all
+classes found within an environment's manifest files.
+
+### Includes filenames (vs. Puppet 4)
+
+Unlike the resource type API under Puppet 4, the environment classes API does include the
+filename in which each class was found. The resource type API under Puppet 3 does include
+the filename but the resource type API under Puppet 4 does not.
+
+### Lacks line numbers
+
+The environment classes API does not include the line number at which a class is found in
+the file.
+
+### Lacks documentation strings (vs. Puppet 3)
+
+Unlike the resource types API under Puppet 3, the environment classes API does not include
+any doc strings for a class entry. Note that doc strings are also not returned for class
+entries in the Puppet 4 resource type API.
+
+### Returns file entries for manifests with no classes
+
+The environment classes API returns a file entry for manifests that exist in the
+environment but in which no classes were found. The resource type API omits entries for
+files which did not contain any classes.
+
+### Uses `application/json` Content-Type
+
+The Content-Type in the response to an environment classes API query is
+`application/json`, whereas the resource types API uses a Content-Type of `text/pson`.
+
+### Includes successfully parsed classes, even if some return errors, and returns error
+messages
```

Comment:
Should the word `messages` be moved to the end of the prior line?  I wasn't sure if it would be preserved in the heading after the docs are published otherwise.